blog/2025-01-23-walkthrough-a-wasmcloud-crud-application-in-go/index.mdx

```diff
@@ -0,0 +1,486 @@
+    wasmcloud.dev/readme-md-url: https://github.com/wasmCloud/go/blob/main/examples/components/http-keyvalue-crud/README.md
+    wasmcloud.dev/homepage: https://github.com/wasmCloud/go/blob/main/examples/components/http-keyvalue-crud
+    wasmcloud.dev/categories: |
+      http,outgoing-http,http-server,tinygo,golang,example
+spec:
+  components:
+  - name: keyvalue-nats
+    type: capability
+    properties:
+      image: ghcr.io/wasmcloud/keyvalue-nats:0.3.1
+    traits: []
+  - name: http-server
+    type: capability
+    properties:
+      image: ghcr.io/wasmcloud/http-server:0.24.0
+    traits:
+    - type: link
+      properties:
+        namespace: wasi
+        package: http
+        interfaces:
+        - incoming-handler
+        source:
+          config:
+          - name: wasi-http-config
+            properties:
+              address: 127.0.0.1:8000
+        target:
+          name: crud
+  - name: crud
+    type: component
+    properties:
+      image: file://./build/http-keyvalue-crud_s.wasm
+      id: crud
+    traits:
+    - type: spreadscaler
+      properties:
+        instances: 100
+    - type: link
+      properties:
+        namespace: wasi
+        package: keyvalue
+        interfaces:
+        - store
+        target:
+          name: keyvalue-nats
+          config:
+          - name: wasi-keyvalue-config
+            properties:
+              bucket: wasmcloud
+              enable_bucket_auto_create: 'true'
+
+```
+
+Let's walk through the manifest.
+
+* The `metadata` fields provide a name, description, and other optional metadata for the application.
+* The `components` fields under `spec` describe the different pieces of our application&mdash;not *just* WebAssembly components, but also capability providers. This application consists of three pieces:
+    - The `keyvalue-nats` capability provider, which mediates between the WebAssembly component we built from `main.go` and the actual key-value store we're using&mdash;in this case, the storage built into the [NATS connective layer](/docs/ecosystem/nats/) that is already part of your wasmCloud environment. This provider is fetched as an [OCI artifact](/docs/concepts/packaging). 
+    - The `http-server` capability provider, which handles HTTP service, also as an OCI artifact.
+    - The `crud` component that we just built, served up straight from the local binary.
+* Under the components, we define configuration and links that connect the entities. Links are defined under the `source` (or importer) on the interface in question. (Learn more about linking on the [Linking at runtime](/docs/concepts/linking-components/linking-at-runtime/) overview.)
+
+When we wrote the code for our WebAssembly component, we didn't worry about how key-value and HTTP services would be rendered. We treated them as abstractions, and now at deployment, we've defined the specific providers that will fulfill those abstract requirements according to the contracts defined in our interfaces. 
```

Comment:
```suggestion
When we wrote the code for our WebAssembly component, we didn't worry about how key-value and HTTP services would be rendered. We treated them as abstractions, and now at deployment, we've defined the specific providers that will fulfill those abstractions with concrete implementations. 
```

Thoughts on this? Trying to cut down on the special words and emphasize abstraction -> concrete.